components/server/src/websocket/websocket-connection-manager.ts

```diff
@@ -428,21 +428,37 @@ class GitpodJsonRpcProxyFactory<T extends object> extends JsonRpcProxyFactory<T>
             observeAPICallsDuration(method, 200, timer());
             return result;
         } catch (e) {
+            const traceID = span.context().toTraceId();
+
             if (e instanceof ResponseError) {
                 increaseApiCallCounter(method, e.code);
                 observeAPICallsDuration(method, e.code, timer());
                 TraceContext.setJsonRPCError(ctx, method, e);
 
-                log.info({ userId }, `Request ${method} unsuccessful: ${e.code}/"${e.message}"`, { method, args });
+                const severityLogger = ErrorCodes.isUserError(e.code) ? log.info : log.error;
+                severityLogger(
+                    { userId },
+                    `JSON RPC Request ${method} failed with user error: ${e.code}/"${e.message}"`,
+                    {
+                        method,
+                        args,
+                        code: e.code,
+                        message: e.message,
+                    },
+                );
             } else {
                 TraceContext.setError(ctx, e); // this is a "real" error
 
-                const err = new ResponseError(ErrorCodes.INTERNAL_SERVER_ERROR, "internal server error");
+                const err = new ResponseError(
+                    ErrorCodes.INTERNAL_SERVER_ERROR,
+                    `Internal server error. Please quote trace ID: '${traceID}' when reaching to Gitpod Support`,
```

Comment:
We attach the trace ID to the error, such that it can be quoted back to us.

We should also always attach the trace in the response headers on all requests, but that's for another PR.

Reply:
I like the direction. But then we need to at least log the traceId as separate field we can query for.

Reply:
In the worst case, we won't have the trace and we're back to where we're today. I agree, we should also attach the trace ID on logs but that's currently difficult, we'd need the following:

* Stop using a global logger
* Instead use a "context" logger - one which is created as part of the request and then used throughout the lifecycle of the request. That way, we can automatically attach the trace ID to the logs in a middleware rather than having to always remember to attach the trace ID.